src/polycap-photon.c

```diff
@@ -348,7 +348,7 @@ polycap_vector3 polycap_photon_pc_intersect(polycap_vector3 photon_coord, polyca
 	phot_temp.y = photon_coord.y + phot_dir.y * (z_fin-photon_coord.z)/phot_dir.z;
 	phot_temp.z = photon_coord.z + phot_dir.z * (z_fin-photon_coord.z)/phot_dir.z;
 
-	return phot_temp;
+	return &phot_temp;
```

Comment:
```suggestion
        polycap_vector3 *rv = malloc(sizeof(polycap_vector3))
        *rv = phot_temp;;
	return rv;
```

Misschien zo iets? Je zou natuurlijk ook gewoon `phot_temp` kunnen malloc'en in het begin van de functie en die dan zo gebruiken.

Vergeet niet te free'en later!